src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect, useState } from 'react';
+import { addComment, deleteComment, getComments } from '../../api/comments';
+import { Comment } from '../../types/Comment';
+import { Post } from '../../types/Post';
+import { Loader } from '../Loader';
+import { NewCommentForm } from '../NewCommentForm/NewCommentForm';
+
+interface Props {
+  post: Post;
+}
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [error, setError] = useState(false);
+  const [message, setMessage] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isWriting, setIsWriting] = useState(false);
+
+  const loadComments = async (postId: number) => {
+    setIsLoading(true);
+    setError(false);
+    setMessage(false);
+    setIsWriting(false);
+    setComments([]);
+
+    try {
+      const loadedComments = await getComments(postId);
+
+      if (loadedComments.length === 0) {
+        setMessage(true);
+      }
+
+      setComments(loadedComments);
+    } catch {
+      setError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadComments(post.id);
+  }, [post]);
+
+  const onDelete = async (commentId: number) => {
+    try {
+      await deleteComment(commentId);
+
+      const newComments = comments.filter(comment => comment.id !== commentId);
+
+      if (newComments.length === 0) {
+        setMessage(true);
+      }
+
+      setComments(newComments);
+    } catch {
+      setError(true);
+    }
+  };
+
+  const onAdd = async ({ name, email, body, id }: Omit<Comment, 'postId'>) => {
+    setIsLoading(true);
+    setMessage(false);
+
+    try {
+      const newComment = await addComment({
+        name,
+        email,
+        body,
+        postId: post.id,
+        id,
+      });
+
+      setComments(prevComments => [...prevComments, newComment]);
+    } catch {
+      setError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <div className="content" data-cy="PostDetails">
+      <div className="content" data-cy="PostDetails">
+        <div className="block">
+          <h2 data-cy="PostTitle">{`#${post.id}: ${post.title}`}</h2>
+
+          <p data-cy="PostBody">{post.body}</p>
+        </div>
+
+        <div className="block">
+          {isLoading && !isWriting ? (
```

Comment:
Ensure that the `isWriting` state is correctly managed to prevent conflicts with the loading state. This will help avoid issues where the loader might be displayed while the user is writing a comment.